### PR TITLE
Recommendation Endpoint

### LINKED ISSUE #3: Make a recommendation
As a user, when I click the button requesting a recommendation, I want to see a cocktail recommendation based on ratings I've provided and ratings from all other users in the system.

### PR DESCRIPTION
**Pull request by mkrumholz**
## Pull request checklist
Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Rubocop has passed locally and any fixes were made for failures
- [ ] CircleCI build is passing
- [ ] All tests passing locally

There are currently 3 failing tests, which I will work on correcting as this gets reviewed. 2 are for validations from the hotfix and 1 is because we've now added an auth layer that needs to be applied more broadly. 

I am planning to submit a new PR on an adjacent branch after running rubocop and getting those tests passing, so no need to worry about those things for now.

## Pull request type
Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: 
closes #3 

## What is the new behavior?

- Updates Factories
- Adds an endpoint for getting a recommendation
- Refactors substantial portions of the recommendation model needed to ensure that this functioned properly
- Tests added

## Does this introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: app/controllers/api/v1/recommendations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class Api::V1::RecommendationsController < ApplicationController
+  before_action :authorize_user
+
+  def show
+    cocktail_id = RecommendationService.recommendation(current_user.id)
+    details = CocktailFacade.retrieve_cocktail(current_user.id, cocktail_id)
+    render json: CocktailDetailsSerializer.details(cocktail_id, details)
+  end
```
Comment (markcyen) [inline app/controllers/api/v1/recommendations_controller.rb:8]: very clean... me likes

File: app/facades/cocktail_facade.rb [hunk 2]
```diff
@@ -1,33 +1,31 @@
 class CocktailFacade
-  def self.retrieve_cocktail(cocktail_id)
+  def self.retrieve_cocktail(user_id, cocktail_id)
     response = CocktailService.get_cocktail_details(cocktail_id)
```
Comment (markcyen) [inline app/facades/cocktail_facade.rb:3]: nice, thanks for refactoring!

File: app/facades/cocktail_facade.rb [hunk 3]
```diff
@@ -1,33 +1,31 @@
...
+      recipe: build_recipe(ingredients, measurements),
       instructions: details[:strInstructions],
-      rating: rating
+      rating: find_rating(user_id, cocktail_id)
     }
+  end
 
-    # CocktailDetails.new(cocktail)
+  def self.find_rating(user_id, cocktail_id)
+    if Rating.find_by(user_id: user_id, cocktail_id: cocktail_id).present?
```
Comment (markcyen) [inline app/facades/cocktail_facade.rb:18]: small note... wonder if it looks cleaner to set this as a variable. as long as it works, works for me!

**Review by markcyen — APPROVED** [review]
sweet... looks good!

File: app/facades/cocktail_facade.rb (hunk 3 above)
Comment (mkrumholz) [reply app/facades/cocktail_facade.rb:18]: I like that! Maybe could also be a method in the Rating model?

Comment (mkrumholz) [reply app/facades/cocktail_facade.rb:18]: Updated!

**Comment by zachjamesgreen** [comment]
The app cant create a user because the google refresh token must be present. taking it out of the validations solves the issue

**Comment by mkrumholz** [comment]
> The app cant create a user because the google refresh token must be present. taking it out of the validations solves the issue

I'll remove that from all of this for now.

**Comment by mkrumholz** [comment]
> > The app cant create a user because the google refresh token must be present. taking it out of the validations solves the issue
> 
> I'll remove that from all of this for now.

@zachjamesgreen also updated to remove the tests for those validations

**Review by Jaybraum — COMMENTED** [review]
Looks good

**Review by Jaybraum — APPROVED** [review]

**Pull request merged**